src/components/structures/InteractiveAuth.js

```diff
@@ -59,6 +59,12 @@ export default React.createClass({
         clientSecret: React.PropTypes.string,
         emailSid: React.PropTypes.string,
 
+        // Callback to be run with the auth object, when it
+        // has been created. This allows the component to
+        // access more specific functions only available on
+        // the auth object itself.
+        onAuthObject: React.PropTypes.func,
```

Comment:
this is surely horrible. why not pass an `opts` object into the `onAuthFinished` callback with the `emailSid` and `clientSecret`?